cmd/zoekt-sourcegraph-indexserver/queue.go

```diff
@@ -252,6 +280,30 @@ func (q *Queue) SetIndexed(opts IndexOptions, state indexState) {
 	item.indexState = state
 	if state != indexStateFail {
 		item.indexed = reflect.DeepEqual(opts, item.opts)
+		item.consecutiveFailures = 0
+		item.backoffUntil = time.Unix(0, 0)
+	} else {
```

Comment:
One of the decisions I made leaning towards keep it simple on mvp design is to strictly rely on `indexStateFail` value triggering the backoff for the given repo. The current logic blanket assigns this for any non nil [error returned](https://sourcegraph.com/github.com/sourcegraph/zoekt/-/blob/cmd/zoekt-sourcegraph-indexserver/main.go?L517) from zoekt-git-index subprocess.

For a more sophisticated decision making: We could take into account the reason for indexing in our current index attempt, and/or the errors returned by preceding indexing attempts that failed, and using this information to determine if we should allow the repo to continue indexing cadence instead of applying backoff. 

For example, if the error varies amongst preceding failures then we could consider the failure to be caused by transient issues and not a fundamental issue that requires intervention.

Reply:
Just relying on this sounds good to me, as you mention we can always incrementally make this more sophisticated.